buildSrc/src/main/groovy/org/apache/beam/gradle/BeamModulePlugin.groovy

```diff
@@ -436,7 +444,7 @@ class BeamModulePlugin implements Plugin<Project> {
     def google_oauth_clients_version = "1.31.0"
     // Try to keep grpc_version consistent with gRPC version in google_cloud_platform_libraries_bom
     def grpc_version = "1.32.2"
-    def guava_version = "25.1-jre"
+    def guava_version = guava25Projects.contains(project.path) ? "25.1-jre" : "30.1-jre"
```

Comment:
We always treat library.java as a global constant. In all existing cases where a project requires a library version that deviates from library.java, we don't use library.java and instead hard-code that dependency in the project's build.gradle. 

IMO making library.java conditional on the project being compiled defeats the purpose of declaring a common version in the first place.